### PR TITLE
Reader Lists: Enable list creation

### PR DESCRIPTION
**Pull request by jsnmoon**
#### Changes proposed in this Pull Request

* Adds a `Create new list` link in the sidebar.
> Screen Shot 2020-11-02 at 3 40 00 PM

* Tweaks data-layer and state handling to work properly with our ListForm.

#### Testing instructions

* Spin this up locally or enable the `reader/list-management` feature flag in the environment of your choice.
* Ensure that you can see the new `Create new list` link in the Reader sidebar. It should be within the Lists dropdown folder.
* Ensure that clicking the link takes you to a working create form.
* Ensure that creating a list takes you directly to the list edit page (`/read/list/:user/:listSlug/edit`).

### CODE REVIEW

**Review by bluefuton — APPROVED** [review]
Lovely. Works well for me 🚢

**Comment by blowery** [comment]
Tests failed due to bad syntax?

```
[22:44:03]Summary of all failing tests
[22:44:03]FAIL state/reader/lists/test/reducer.js
[22:44:03]  ● Test suite failed to run
[22:44:03]
[22:44:03]    Jest encountered an unexpected token
[22:44:03]
[22:44:03]    This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript.
[22:44:03]
[22:44:03]    By default, if Jest sees a Babel config, it will use that to transform your files, ignoring "node_modules".
[22:44:03]
[22:44:03]    Here's what you can do:
[22:44:03]     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
[22:44:03]     • If you need a custom transformation specify a "transform" option in your config.
[22:44:03]     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.
[22:44:03]
[22:44:03]    You'll find more details and examples of these config options in the docs:
[22:44:03]    https://jestjs.io/docs/en/configuration.html
[22:44:03]
[22:44:03]    Details:
[22:44:03]
[22:44:03]    SyntaxError: /home/teamcity-2/buildAgent/work/c4a9d5b38c1dacde/client/state/reader/lists/test/reducer.js: Unexpected token (289:0)
[22:44:03]
[22:44:03]      287 | 	} );
[22:44:03]      288 | } );
[22:44:03]    > 289 | 
[22:44:03]          | ^

```

File: client/state/reader/lists/test/reducer.js [hunk 1]
```diff
@@ -274,6 +273,16 @@ describe( 'reducer', () => {
 					listId: 1,
 				} )
 			).toEqual( [ 2 ] );
+		test( 'should add a list on creation', () => {
```
Comment (blowery) [inline client/state/reader/lists/test/reducer.js:276]: forgot to close the previous test here

Comment (blowery) [inline client/state/reader/lists/test/reducer.js:276]:
```suggestion
		} );
		
		test( 'should add a list on creation', () => {
```

**Review by blowery — CHANGES_REQUESTED** [review]
fix the test, then it's gtg

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5110585

Thank you @jsnmoon for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5110585).

**Pull request merged**